service/api/client/client.go

```diff
@@ -0,0 +1,282 @@
+package client
+
+import (
+	"context"
+	"io"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+
+	"github.com/ipfs/go-cid"
+	cbornode "github.com/ipfs/go-ipld-cbor"
+	format "github.com/ipfs/go-ipld-format"
+	logging "github.com/ipfs/go-log"
+	"github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/textileio/go-threads/cbor"
+	core "github.com/textileio/go-threads/core/service"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/go-threads/crypto/symmetric"
+	"github.com/textileio/go-threads/service"
+	pb "github.com/textileio/go-threads/service/api/pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var log = logging.Logger("threadserviceclient")
+
+// Client provides the client api.
+type Client struct {
+	c    pb.APIClient
+	conn *grpc.ClientConn
+}
+
+var _ core.API = (*Client)(nil)
+
+// NewClient starts the client.
+func NewClient(target string, opts ...grpc.DialOption) (*Client, error) {
+	conn, err := grpc.Dial(target, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		c:    pb.NewAPIClient(conn),
+		conn: conn,
+	}, nil
+}
+
+// Close closes the client's grpc connection and cancels any active requests.
+func (c *Client) Close() error {
+	return c.conn.Close()
+}
+
+// GetHostID returns the service's host peer ID.
+func (c *Client) GetHostID(ctx context.Context) (peer.ID, error) {
+	resp, err := c.c.GetHostID(ctx, &pb.GetHostIDRequest{})
+	if err != nil {
+		return "", err
+	}
+	return peer.IDFromBytes(resp.PeerID)
+}
+
+func (c *Client) CreateThread(ctx context.Context, id thread.ID) (info thread.Info, err error) {
+	resp, err := c.c.CreateThread(ctx, &pb.CreateThreadRequest{
+		ThreadID: id.Bytes(),
+	})
+	if err != nil {
+		return
+	}
+	return threadInfoFromPbThread(resp)
+}
+
+func (c *Client) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.AddOption) (info thread.Info, err error) {
+	args := &core.AddOptions{}
+	for _, opt := range opts {
+		opt(args)
+	}
+	resp, err := c.c.AddThread(ctx, &pb.AddThreadRequest{
+		Addr:      addr.Bytes(),
+		ReadKey:   args.ReadKey.Bytes(),
```

Comment:
AddThread is for adding an existing thread... this can be done without the read key. I think what you want is a modification / addition to create thread. Maybe we can change it so CreateThread takes keys instead of making them internally.

Reply:
I see, yes that makes sense. And yes, this is indeed needed. If CreateThread could take keys directly, this would solve that use case of keeping keys 'local' 👍

Reply:
Ok, cool. I think we'll also need an additional method to add an already encrypted event. Maybe that's what AddRecord should be and a new CreateRecord becomes the old AddRecord.

Reply:
Awesome, agree re: names. AddRecord for adding existing (encrypted/structured) Record, CreateRecord for taking 'raw' input and bundling it up automagically.

Reply:
So, 
```
AddRecord(ctx context.Context, id thread.ID, event Event) (ThreadRecord, error)
```
That way, the client doesn't need to get the prevID and sign the record. They just give the encrypted event. Sound okay?

Reply:
Excellent! Yup makes sense. Actually, this is fantastic, because the signature stuff is probs the most tricky piece of the whole dance in the browser.

Reply:
Ok, cool. Though, thinking about it some more, maybe you do want to handle that piece? That way the service doesn't need the private key? I could see it either way.

Reply:
Ah right, duh... no you're right... we don't want the service to have anything beyond the follower key, that's the whole point of the JS stuff really. So we'll still need to do the signature stuff in browser. That's _fine_, the code is already there to do that.